src/main/java/io/r2dbc/postgresql/util/Assert.java

```diff
@@ -147,4 +132,44 @@ public static void isTrue(boolean expression, String message) {
         }
     }
 
+    /**
+     * Checks that the provided URL exists or null.
+     *
+     * @param path    path to check
+     * @param message the message to use in exception if type is not as required
+     * @return existing url
+     * @throws IllegalArgumentException if {@code value} is not of the required type
+     * @throws IllegalArgumentException if {@code value}, {@code type}, or {@code message} is {@code null}
+     */
+    public static URL requireUrlExistsOrNull(@Nullable String path, String message) {
```

Comment:
We should revert changes to the `Assert` class and instead, introduce rather an utility method to figure out, whether the given `ConnectionFactoryOption` tries to point to a real file or to a URL. Checking for a scheme would be a good approach if the configured string doesn't point to a file.

We might still run into an ambiguity on windows (`c:\foo\bar.crt` vs. `classpath:foo.crt` or `classpath:/foo`). Not sure how to resolve that.